ui/hooks/useDisplayName.ts

```diff
@@ -9,22 +9,24 @@ import { useName } from './useName';
  * @param value
  * @param type
  * @param variation
- * @returns The name if it can be resolved, otherwise null.
+ * @returns An object with two properties:
+ * - `name` {string|null} - The display name, if it can be resolved, otherwise null.
+ * - `hasPetname` {boolean} - True iff there is a petname for the given address.
  */
 export function useDisplayName(
   value: string,
   type: NameType,
   variation?: string,
-): string | null {
+): { name: string | null; hasPetname: boolean } {
   const nameEntry = useName(value, type, variation);
   const contractName = useSelector((state) =>
     (getMemoizedMetadataContractName as any)(state, value),
   );
   if (nameEntry?.name) {
-    return nameEntry.name;
+    return { name: nameEntry.name, hasPetname: true };
   }
   if (contractName) {
-    return contractName;
+    return { name: contractName, hasPetname: false };
```

Comment:
Not worth a refactor now, but if we ever support additional display name sources in future, it could be more scalable to have the second property be a new enum such as `DisplayNameType` so we can track a single `name` but also where it came from.

Reply:
Yes, that's a good point. We should address that when we make [the next version of name resolution](https://www.notion.so/metamask-design-system/Petnames-Displaying-Tokens-9e38dc56ceac44c0aacf65eabb8dfcca?pvs=4#e1c6ee4a98a94ca885c472b7f9a66b95).